### PR TITLE
Add ModelInferenceConfig type

### PR DESCRIPTION
**Pull request by SIGLOST**

### CODE REVIEW

File: lumeo_api_client/src/models.rs [hunk 1]
```diff
@@ -15,15 +15,43 @@ pub struct Model {
     pub application_id: Option<Uuid>,
     pub name: String,
     pub description: Option<String>,
+    pub weights_filename: Option<String>,
     pub weights_file_url: String,
+    pub metadata_filename: Option<String>,
```
Comment (FSMaxB) [inline lumeo_api_client/src/models.rs:20]: Naive question because I don't really know how this is used, but wouldn't the filename be part of the URL?

**Review by FSMaxB — APPROVED** [review]
Can't really comment a lot on the properties themselves.

File: lumeo_api_client/src/models.rs (hunk 1 above)
Comment (SIGLOST) [reply lumeo_api_client/src/models.rs:20]: Yes the filename is part of the URL, I was performing the url string parsing before but besides the initial part of the URL it also includes several tokens after the filename. `?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=V5P2RSENIP..."`
So the string parsing does not looked to be a very robust solution if for some reason we change the URL format in the future. 
Maybe there's some good rust crate for parsing complex URLs? 

Basically `lumeod` is responsible to download those model's files (weights, metadata, labels) to device folder `/var/lib/lumeo/models/`.
`lumeod` stores them with a different filename using the format `<model_uuid>.original_file_extension`. And we need to include those filenames in the inference config file we generate.

Comment (SIGLOST) [reply lumeo_api_client/src/models.rs]: Thank you. Yes AFAIK currently the filenames info is only relevant to `lumeod`. 
I've removed the fields here on `api-client` and we're parsing of the URLs on the `lumeod` PR

**Review by FSMaxB — APPROVED** [review]

File: lumeo_api_client/src/models.rs [hunk 2]
```diff
@@ -67,6 +92,60 @@ pub enum Format {
...
+    pub nms_iou_threshold: Option<f64>,
+
+    pub object_min_size: Option<Resolution>,
+    pub object_max_size: Option<Resolution>,
+
+    pub top_k: Option<i32>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
```
Comment (DmitrySamoylov) [inline lumeo_api_client/src/models.rs:142]: Let's use `snake_case` everywhere.

File: lumeo_api_client/src/models.rs [hunk 3]
```diff
@@ -20,6 +20,7 @@ pub struct Model {
     pub metadata_file_url: Option<String>,
     pub parameters: BTreeMap<String, String>,
     pub gallery_img_url: Option<String>,
+    pub model_inference_config: Option<ModelInferenceConfig>,
```
Comment (DmitrySamoylov) [inline lumeo_api_client/src/models.rs:23]: Since it's a field in `Model` then maybe we can reduce it to `inference_config`?

File: lumeo_api_client/src/models.rs [hunk 4]
```diff
@@ -18,6 +17,7 @@ pub struct Model {
     pub description: Option<String>,
     pub weights_file_url: String,
     pub metadata_file_url: Option<String>,
+    pub labels_file_url: Option<String>,
```
Comment (DmitrySamoylov) [inline lumeo_api_client/src/models.rs:20]: Do we want to upload labels to s3 or we can include them directly as a `Vec<String>`?

File: lumeo_api_client/src/models.rs [hunk 5]
```diff
@@ -24,6 +25,30 @@ pub struct Model {
...
 
+#[derive(Debug, Deserialize)]
+pub struct ModelInferenceConfig {
+    pub net_scale_factor: f64,
+    pub color_format: ModelColorFormat,
+    pub network_mode: ModelNetworkMode,
+    pub infer_dims: Option<String>,
+    pub input_order: Option<ModelInputOrder>,
+    pub input_blob_name: Option<String>,
+    pub output_blob_names: Option<String>,
```
[the remark this answers is not in the document]
Comment (SIGLOST) [reply lumeo_api_client/src/models.rs]: Yeah, in DS config we load a single string in the format `output1;output2;output3` but here it should be a vector.

File: lumeo_api_client/src/models.rs (hunk 2 above)
Comment (SIGLOST) [reply lumeo_api_client/src/models.rs]: Do you mean in all the other enum of this file or just in the ones I've added?

File: lumeo_api_client/src/models.rs (hunk 4 above)
Comment (SIGLOST) [reply lumeo_api_client/src/models.rs:20]: Discussed it a bit with Patricia, and she wants to handle on Console both: file upload and text input (that will be also converted to file). 
I'm not sure if in some cases might be expensive to the payload pass it as vector, but we can adjust later.

**Review by DmitrySamoylov — APPROVED** [review]
